http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/MonitorAtBoot.scala

```diff
@@ -157,61 +177,148 @@ class MonitorAtBoot[F[_]](publisherQueue: Queue[F, LeoPubsubMessage],
                   )
                 )
             }
-            googleProject <- F.fromOption(
-              LeoLenses.cloudContextToGoogleProject.get(cluster.cloudContext),
-              new RuntimeException(
-                "trying to provision app on Azure during MonitorAtBoot. This is not supported yet"
-              ) // TODO: support azure
-            )
-            machineType <- computeService
-              .getMachineType(
-                googleProject,
-                ZoneName("us-central1-a"),
-                nodepool.machineType
-              ) // TODO: if use non `us-central1-a` zone for galaxy, this needs to be udpated
-              .flatMap(opt =>
-                F.fromOption(
-                  opt,
-                  new LeoException(s"can't find machine config for ${app.appName.value}", traceId = Some(traceId))
-                )
-              )
+            msg <- cluster.cloudContext match {
+              case CloudContext.Gcp(googleProject) =>
+                for {
+                  machineType <- computeService
+                    .getMachineType(
+                      googleProject,
+                      ZoneName("us-central1-a"),
+                      nodepool.machineType
+                    ) // TODO: if use non `us-central1-a` zone for galaxy, this needs to be udpated
+                    .flatMap(opt =>
+                      F.fromOption(
+                        opt,
+                        new LeoException(s"can't find machine config for ${app.appName.value}",
+                                         traceId = Some(appContext.traceId)
+                        )
+                      )
+                    )
+
+                  diskIdOpt = app.appResources.disk.flatMap(d =>
+                    if (d.status == DiskStatus.Creating) Some(d.id) else None
+                  )
+                  msg = CreateAppMessage(
+                    googleProject,
+                    action,
+                    app.id,
+                    app.appName,
+                    diskIdOpt,
+                    app.customEnvironmentVariables,
+                    app.appType,
+                    app.appResources.namespace.name,
+                    Some(AppMachineType(machineType.getMemoryMb / 1024, machineType.getGuestCpus)),
+                    Some(appContext.traceId)
+                  )
+                } yield msg
+
+              case CloudContext.Azure(_) =>
+                for {
+                  tokenOpt <- samDAO.getCachedArbitraryPetAccessToken(app.auditInfo.creator)
+                  userToken <- F.fromOption(
+                    tokenOpt,
+                    MonitorAtBootException(
+                      s"Unable to get pet token for ${app.auditInfo.creator} for app ${app.id} in Provisioning status",
+                      appContext.traceId
+                    )
+                  )
+                  workspaceId: WorkspaceId = app.workspaceId.getOrElse(
```

Comment:
Can we use `F.fromOption` here (and in places where we need to throw exceptions)?